aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/UpdateHandler.java

```diff
@@ -152,6 +152,30 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                                 updateAssociatedRoles(proxy, rdsClient, progress, previousRoles, desiredRoles),
                         CallbackContext::isUpdatedRoles, CallbackContext::setUpdatedRoles)
                 )
+                .then(progress -> Commons.execOnce(progress, () -> {
+                    if ((ResourceModelHelper.shouldStopAutomaticBackupReplication(request.getPreviousResourceState(), request.getDesiredResourceState())
+                            || ResourceModelHelper.shouldStartAutomaticBackupReplication(request.getPreviousResourceState(), request.getDesiredResourceState()))
+                            && StringUtils.isNullOrEmpty(callbackContext.getDbInstanceArn())) {
+                        final DBInstance dbInstance = fetchDBInstance(rdsProxyClient.defaultClient(), progress.getResourceModel());
+                        callbackContext.setDbInstanceArn(dbInstance.dbInstanceArn());
+                    }
+                    return progress;
+                    },  (m) -> !StringUtils.isNullOrEmpty(callbackContext.getDbInstanceArn()), (v, c) -> {}))
```

Comment:
Same comment as earlier: we need to absorb it in a single place. Fetching an instance is a relatively cheap operation, hence we can do it unconditionally if needed.